add-on/src/lib/common.js

```diff
@@ -383,17 +414,60 @@ function updateBrowserActionBadge () {
     badgeColor = '#8C8C8C'
     badgeIcon = '/icons/ipfs-logo-off.svg'
   }
-  return setBrowserActionBadge(badgeText, badgeColor, badgeIcon)
+  try {
+    await browser.browserAction.setBadgeBackgroundColor({color: badgeColor})
+    await browser.browserAction.setBadgeText({text: badgeText})
+    await setBrowserActionIcon(badgeIcon)
+  } catch (error) {
+    console.error('Unable to update browserAction badge due to error', error)
+  }
 }
 
-function setBrowserActionBadge (text, color, icon) {
-  return Promise.all([
-    browser.browserAction.setBadgeBackgroundColor({color: color}),
-    browser.browserAction.setBadgeText({text: text}),
-    browser.browserAction.setIcon({path: icon})
-  ])
+async function setBrowserActionIcon (iconPath) {
+  let iconDefinition = {path: iconPath}
+  try {
+    // Try SVG first -- Firefox supports it natively
+    await browser.browserAction.setIcon(iconDefinition)
+  } catch (error) {
+    // Fallback!
+    // Chromium does not support SVG [ticket below is 8 years old, I can't even..]
+    // https://bugs.chromium.org/p/chromium/issues/detail?id=29683
+    // Still, we want icon, so we precompute rasters of popular sizes and use them instead
+    await browser.browserAction.setIcon(rasterIconDefinition(iconPath))
+  }
 }
 
+function rasterIconDefinition (svgPath) {
+  // icon sizes to cover ranges from:
+  // - https://bugs.chromium.org/p/chromium/issues/detail?id=647182
+  // - https://developer.chrome.com/extensions/manifest/icons
+  return {
+    'path': {
+      '19': rasterIconPath(svgPath, 19),
+      '38': rasterIconPath(svgPath, 38),
+      '128': rasterIconPath(svgPath, 128)
+    }
+  }
+}
+
+function rasterIconPath (iconPath, size) {
+  // point at precomputed PNG file
+  let baseName = /\/icons\/(.+)\.svg/.exec(iconPath)[1]
+  return `/icons/png/${baseName}_${size}.png`
+}
+
+/* Easter-Egg: PoC that generates raster on the fly ;-)
+function rasterIconData (iconPath, size) {
+  let icon = new Image()
+  icon.src = iconPath
+  let canvas = document.createElement('canvas')
+  let context = canvas.getContext('2d')
+  context.clearRect(0, 0, size, size)
+  context.drawImage(icon, 0, 0, size, size)
+  return context.getImageData(0, 0, size, size)
+}
+*/
+
 // OPTIONS
 // ===================================================================
```

Comment:
Yes, at some point. 
I've added those "section" comments, as `common.js` grew organically.  

I was not sure what is the best course of action for "webextension modularization", as Firefox WX ecosystem was still brand new when I started the rewrite from legacy SDK.

In Chrome world people use [ES6 modules](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import), but feature is not implemented in any browsers natively at this time and requires you to transpile everything to a single `.js` file ([example](https://github.com/samuelsimoes/chrome-extension-webpack-boilerplate)).

I avoided it and right now add-on code runs "as-is" (no transpilling), but I am not sure if this approach can be maintained in the long run, especially now that we support multiple browser vendors.

If you feel it is the time, create a new Issue and lcopy/link this comment there.
Alternatively, we leave it be a monolith until ES6 modules are natively supported by both Firefox and Chrome.